### PR TITLE
chore(AclFamily): Move some of the regressions tests to unit and add

### PR DESCRIPTION
**Pull request by kostasrim**
* Move error handling tests from regression to unit
* Move ACL LIST regression to unit test
* Move AUTH regression to unit test
* Move ACL WHOAMI regression to unit test
* Add unit tests for SETUSER/DELUSER (so they run on every PR)
* Add unit tests for all ACL categories

Some of the ACL SETUSER/DELUSER tests could be moved from regressions to unit. I did not want to do that as 
SETUSER/DELUSER also streamline the updates to all proactors (via the MainListener) which in the case of unit tests
this is empty... Therefore, for these we got both, units such that they run on every PR, and regressions to check the full execution path not covered by units.

### CODE REVIEW

File: src/server/acl/acl_family.cc [hunk 1]
```diff
@@ -149,7 +149,7 @@ std::variant<User::UpdateRequest, ErrorReply> ParseAclSetUser(CmdArgList args) {
 
     auto [cat, add] = MaybeParseAclCategory(command);
     if (!cat) {
-      return ErrorReply(absl::StrCat("Unrecognized parameter", command));
+      return ErrorReply(absl::StrCat("Unrecognized parameter ", command));
```
Comment (kostasrim) [inline src/server/acl/acl_family.cc:152]: That was a bug!

File: src/server/acl/acl_family_test.cc [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    }
+  }
+
+  for (auto& cat : acl::REVERSE_CATEGORY_INDEX_TABLE) {
+    if (cat != "_RESERVED") {
+      auto resp = Run({"ACL", "SETUSER", "kostas", absl::StrCat("+@", cat)});
+      EXPECT_THAT(resp, "OK");
+    }
+  }
+  // This won't work because of __RESERVED
```
Comment (kostasrim) [inline src/server/acl/acl_family_test.cc:143]: Will fix these on a separate PR

**Comment by dranikpg** [comment]
Now that we'll soon run pytests on PRs the transition  isnot that important anymore. However unit tests are still faster than pytests and so they take up less time

File: src/server/acl/acl_family_test.cc [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+#include "base/logging.h"
+#include "facade/facade_test.h"
+#include "server/acl/acl_commands_def.h"
+#include "server/command_registry.h"
+#include "server/test_utils.h"
+
+using namespace testing;
+using namespace std;
+using namespace util;
+using namespace boost;
```
Comment (dranikpg) [inline src/server/acl/acl_family_test.cc:18]: What's that for 🤨

**Review by dranikpg — APPROVED** [review]

File: src/server/acl/acl_family_test.cc (hunk 3 above)
Comment (dranikpg) [reply src/server/acl/acl_family_test.cc:18]: Ah I always fall for your auto merge 😆

Comment (kostasrim) [reply src/server/acl/acl_family_test.cc:18]: @dranikpg it's a trap! 

I will remove this, hold!

**Pull request merged**